src/detect-tag-pcap.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright (C) 2007-2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ *  This adds support for creating .pcap output files for each tagged flow.
+ *  File information is stored with the tag, rather than the flow. The reasons
+ *  for this are multi-fold:
+ *  1) Flows are locking on changes while tags are not.
+ *  2) A flow may have multiple tags.
+ *  3) Tags know when to stop following the flow (Thus when to close the file)
+ *  4) A flow id can be used to lookup a flow's tags (Trivial and non-locking).
+ */
+#ifndef __DETECT_TAG_PCAP_H__
+#define __DETECT_TAG_PCAP_H__
+
+#include "suricata-common.h"
+#include "suricata.h"
+#include "decode.h"
+#include "detect.h"
+#include "log-pcap-stream.h"
+#include "pcap.h"
+#include "stream-tcp-private.h"
+
+#define PCAP_PATH_MAX 512
+
+typedef struct TaggedPcapEntry_ {
+    pcap_dumper_t *pcap_dumper;
+    pcap_t *pcap_dead_handle;
+    char pcap_file_path[PCAP_PATH_MAX];
```

Comment:
AFAICS this is really only used in cleanup, not per packet. I wonder, can we just reconstruct the path then to avoid storing this here? Then we'd just store the much smaller things needed to construct the path here?

Reply:
This makes sense to me. Most of the length in that file name has to do with the output_directory and the potential of it being very long. This is already a global variable in detect-tag-pcap.c. I'll make this adjustment so that the TaggedPcapEntry would only need to contain the things necessary for the filename itself, rather than the absolute path.